metrics-operator/controllers/common/analysis/objective_evaluator.go

```diff
@@ -0,0 +1,64 @@
+package analysis
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+)
+
+type ObjectiveEvaluator struct {
+	TargetEvaluator ITargetEvaluator
+}
+
+func NewObjectiveEvaluator(t ITargetEvaluator) ObjectiveEvaluator {
+	return ObjectiveEvaluator{
+		TargetEvaluator: t,
+	}
+}
+
+func (oe *ObjectiveEvaluator) Evaluate(values map[string]string, obj v1alpha3.Objective) types.ObjectiveResult {
```

Comment:
in my understanding an objective evaluator should evaluate the target for a single value, why do we need to pass a  map  of multiple values here? I would much more prefer a signature like this Evaluate(value string, obj Objective)

Reply:
We are passing the map of values due to that we want to store the possible error (cannot retrieve value) inside the objective result and not one layer above (there it would be chaotic). 

Let's think about a scenario where we are trying to evaluate an objective, which (from some unknown reason) does not have a result value. If we try to retrieve this in the analysisEvaluator -> we get an error, but we do not have an objectiveResult, where we can store it -> therefore analysisEvaluator needs to create ObjectiveResult -> this is bad practice, as in this case we do not have a clear separation of roles (analysisEvaluator should create only analysisResults, delegate the rest of the actions to objectiveEvaluator, just store the results and compute the scores).

Reply:
but this logic should probably be part of the layer above or a separate function, anyhow you can return a result with objective info and errror when calling  getValueFromMap

Reply:
but this result objective result needs to be formed somewhere and that should be in objectiveEvaluator